rmw_connext_shared_cpp/src/event.cpp

```diff
@@ -0,0 +1,125 @@
+// Copyright 2015-2017 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+#include "rmw/error_handling.h"
+#include "rmw/impl/cpp/macros.hpp"
+
+#include "rmw_connext_shared_cpp/types.hpp"
+#include "rmw_connext_shared_cpp/event.hpp"
+#include "rmw_connext_shared_cpp/event_converter.hpp"
+#include "rmw_connext_shared_cpp/connext_static_event_info.hpp"
+
+
+rmw_event_t *
+__rmw_create_publisher_event(
+  const char * implementation_identifier,
+  const rmw_publisher_t * publisher,
+  const rmw_event_type_t event_type)
+{
+  if (publisher->implementation_identifier != implementation_identifier) {
+    RMW_SET_ERROR_MSG("client handle not from this implementation");
+    return nullptr;
+  }
+
+  rmw_event_t * rmw_event = rmw_event_allocate();
+  if (nullptr == rmw_event) {
+    RMW_SET_ERROR_MSG("failed to allocate subscription");
+    return nullptr;
+  }
+
+  rmw_event->implementation_identifier = publisher->implementation_identifier;
+  rmw_event->data = static_cast<ConnextCustomEventInfo *>(publisher->data);
+  rmw_event->event_type = event_type;
+
+  return rmw_event;
+}
+
+
+rmw_event_t *
+__rmw_create_subscription_event(
+  const char * implementation_identifier,
+  const rmw_subscription_t * subscription,
+  const rmw_event_type_t event_type)
+{
+  if (subscription->implementation_identifier != implementation_identifier) {
+    RMW_SET_ERROR_MSG("client handle not from this implementation");
+    return nullptr;
+  }
+
+  rmw_event_t * rmw_event = rmw_event_allocate();
+  if (nullptr == rmw_event) {
+    RMW_SET_ERROR_MSG("failed to allocate subscription");
+    return nullptr;
+  }
+
+  rmw_event->implementation_identifier = subscription->implementation_identifier;
+  rmw_event->data = static_cast<ConnextCustomEventInfo *>(subscription->data);
```

Comment:
We could remove this by having subscription have a GetCustomeEventInfo()

Reply:
This looks like an old comment compared to the code I'm currently viewing.